### PR TITLE
Rebranding user facing DLM references

### PR DESCRIPTION
**Pull request by gmarouli**
In this PR we change the code references to DLM to fit the new branding as data stream lifecycle. This change affects:

- [x] The cluster settings
- [x] The explain lifecycle API

Part of: https://github.com/elastic/elasticsearch/issues/96875
Relates to: https://github.com/elastic/elasticsearch/issues/93596

### CODE REVIEW

**Comment by gmarouli** [comment]
@elasticmachine update branch

File: docs/reference/dlm/apis/explain-data-lifecycle.asciidoc [hunk 1]
```diff
@@ -55,7 +55,7 @@ GET .ds-metrics-2023.03.22-000001/_lifecycle/explain
 --------------------------------------------------
 // TEST[skip:we're not setting up DLM in these tests]
 
-If the index is managed by DLM `explain` will show the `managed_by_dlm` field
+If the index is managed by DLM `explain` will show the `managed_by_lifecycle` field
```
Comment (andreidan) [inline docs/reference/dlm/apis/explain-data-lifecycle.asciidoc:58]:
```suggestion
If the index is managed by the data stream lifecycle `explain` will show the `managed_by_lifecycle` field
```

Comment (andreidan) [reply docs/reference/dlm/apis/explain-data-lifecycle.asciidoc:58]: I believe this file contains more references to dlm - https://github.com/elastic/elasticsearch/blob/main/docs/reference/dlm/apis/explain-data-lifecycle.asciidoc?plain=1#L27
https://github.com/elastic/elasticsearch/blob/main/docs/reference/dlm/apis/explain-data-lifecycle.asciidoc?plain=1#L58

Comment (andreidan) [reply docs/reference/dlm/apis/explain-data-lifecycle.asciidoc:58]: @n1v0lg Can we rename the `manage_dlm` privilege to `manage_lifecycle` or so? https://github.com/elastic/elasticsearch/blob/main/docs/reference/dlm/apis/explain-data-lifecycle.asciidoc?plain=1#L16

Are there BWC concerns?

Comment (andreidan) [reply docs/reference/dlm/apis/explain-data-lifecycle.asciidoc:58]: The security rebranding could be done in a follow-up

**Review by andreidan — APPROVED** [review]
LGTM, thanks for working on this Mary

File: docs/reference/dlm/apis/explain-data-lifecycle.asciidoc (hunk 1 above)
Comment (gmarouli) [reply docs/reference/dlm/apis/explain-data-lifecycle.asciidoc:58]: I was planning to address this in the [documentation PR](https://github.com/elastic/elasticsearch/pull/95326), in order to ensure I will be using consistent language in the tutorials and the APIs. Would that be ok?

Comment (gmarouli) [reply docs/reference/dlm/apis/explain-data-lifecycle.asciidoc:58]: About rebranding security I added it as an extra task in https://github.com/elastic/elasticsearch/issues/96875

Comment (n1v0lg) [reply docs/reference/dlm/apis/explain-data-lifecycle.asciidoc:58]: I think we should be able to do this without BWC concerns: we introduced [the new privilege in 8.9](https://github.com/elastic/elasticsearch/pull/95512), and haven't removed the DLM feature flag yet, so changing it in 8.9 should be fine, without any extra work needed.

We just have to make sure we update raw string references to `manage_dlm` everywhere to the new privilege name. @gmarouli will you be able to pick this up?

Comment (gmarouli) [reply docs/reference/dlm/apis/explain-data-lifecycle.asciidoc:58]: Yes let me do this right away, now that it's fresh in our minds :)

**Pull request merged**